src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -226,14 +246,17 @@ private static bool Compile(ProjectContext context, string configuration, string
                 foreach (var projectDependency in Sort(projects))
                 {
                     // Skip compiling project dependencies since we've already figured out the build order
-                    var compileResult = Command.Create("dotnet-compile", 
+                    var compileResult = ResolveCommand("dotnet-compile", 
                         $"--framework {projectDependency.Framework} " +
                         $"--configuration {configuration} " +
                         $"--output \"{outputPath}\" " +
                         $"--temp-output \"{intermediateOutputPath}\" " +
                         "--no-project-dependencies " +
                         (noHost ? "--no-host " : string.Empty) +
-                        $"\"{projectDependency.Project.ProjectDirectory}\"")
+                        (profile ? BuildProfile + " " : string.Empty) +
```

Comment:
In the current PR, each project project reasons about preconditions in isolation (detects its own, prints its own). Therefore I had to propagate the flags to dependency invocations as well.

This will go away once I switch to a centralized precondition checking that collects preconditions for all projects.